app/models/payment.rb

```diff
@@ -32,34 +32,35 @@ class Payment < ApplicationRecord
   belongs_to :user
 
   scope :current_camp_payments, -> { where('camp_year = ? ', CampConfiguration.active_camp_year) }
-  scope :status1_current_camp_payments, -> { current_camp_payments.where('transaction_status = ?', "1") }
+  scope :status1_current_camp_payments, -> { current_camp_payments.where('transaction_status = ?', '1') }
 
   private
 
   def set_status
-    return unless self.transaction_status == '1'
+    return unless transaction_status == '1'
 
-    @current_enrollment = self.user.enrollments.current_camp_year_applications.last
-    if self.user.payments.current_camp_payments.where(transaction_status: 1).count == 1
-      RegistrationMailer.app_complete_email(self.user).deliver_now
-      @current_enrollment.update!(application_status: "submitted", application_status_updated_on: Date.today)
-      if @current_enrollment.recommendation.present?
-        if @current_enrollment.recommendation.recupload.present? 
-          @current_enrollment.update!(application_status: "application complete", application_status_updated_on: Date.today)
-        end
-      end
-    else 
-      if balance_due == 0 && @current_enrollment.camp_doc_form_completed
-        @current_enrollment.update!(application_status: "enrolled", application_status_updated_on: Date.today)
+    @current_enrollment = user.enrollments.current_camp_year_applications.last
+
+    # Check if this is the first payment (when required)
+    if @current_enrollment.application_fee_required && user.payments.current_camp_payments.where(transaction_status: 1).count == 1
```

Comment:
status1_current_camp_payments scope might be used in line 45

Reply:
Not here due to the specificity of the payment count